src/sentry/static/sentry/app/views/settings/organizationTeams/teamDetails.jsx

```diff
@@ -87,7 +133,26 @@ const TeamDetails = createReactClass({
 
     if (this.state.loading) {
       return <LoadingIndicator />;
-    } else if (!team || this.state.error) {
+    } else if (!team || !team.hasAccess) {
```

Comment:
Treat this differently than if there is error state (e.g. network error) vs no permissions.